### DISCUSSION TITLE
ILI9431 using Tasmota32-lvgl 12.5.0 does not initialise correctly after hard boot

### DISCUSSION THREAD

**Discussion by RichardLaxton** [arendst/Tasmota#18924] [Support request]
I have one of the common ILI9341 320x240 screens as has been discussed a lot here:

![210751832-7e576ca8-7ab3-423c-84ca-fd9f2a8ea58b](https://github.com/arendst/Tasmota/assets/21622875/af507a53-72d9-468e-a165-617768f8125a)

I am using the standard 12.5.0 image installed using the web installer.

I have managed to get the screen working very well, but only if I set the display.ini contents to the standard ST7789_display.ini then set the display.ini to the correct one (shown below) for my ILI9341. If I hard reset or power on boot with the ILI9341 display.ini then I just get a white screen until I swap in the ST7789 file, reboot, then swap back to the ILI9341 and soft boot.

Anyway, my ESP32 dev kit is configured as follows:
<img width="231" alt="Screenshot 2023-06-21 at 8 14 55 pm" src="https://github.com/arendst/Tasmota/assets/21622875/1b739451-2c41-4b7b-8927-c0a5e1645dc7">

Both the screen and the XPT2046 are on the same VSPI bus and CS are set accordingly. The problem described happens even if all connections and configuration for the touch screen is dropped. There is an LED attached to the PWM output that is not connected to the screen.

The display.ini file that I am using is:
```
:H,ILI9341,240,320,16,SPI,1,*,*,*,*,*,*,*,40
:S,2,1,1,0,140,20
:I
:B,60,1
:r,3
EF,3,03,80,02
CF,3,00,C1,30
ED,4,64,03,12,81
E8,3,85,00,78
CB,5,39,2C,00,34,02
F7,1,20
EA,2,00,00
C0,1,23
C1,1,10
C5,2,3e,28
C7,1,86
36,1,48
37,1,00
3A,1,55
B1,2,00,18
B6,3,08,82,27
F2,1,00
26,1,01
E0,0F,0F,31,2B,0C,0E,08,4E,F1,37,07,10,03,0E,09,00
E1,0F,00,0E,14,03,11,07,31,C1,48,08,0F,0C,31,36,0F
11,80
29,80
:o,28
:O,29
:A,2A,2B,2C,16
:R,36
:0,48,00,00,80
:1,28,00,00,81
:2,88,00,00,82
:3,E8,00,00,83
:i,20,21
:TS,*,14,0
:M,350,3890,3898,281
#
``` 

It feels like the ILI9341 driver is not fully initialising the screen, while the ST7789 (which does sort of display an image, albeit with the wrong resolution and colours) is fully initialising the screen so that until I do a hard reset, the screen will work. As soon as I do hard reset, the LCD module is being half initialised. My other thought was a race condition in the way that the RESET, CS etc pins where being set, but that does not make much sense since a normal soft reset works fine to initialise the screen.

**Comment by sfromis** [comment]
I see that you did not include a `Display Rst` gpio in your config, and you did not specify what you're doing (or not) with the RESET pin available on the board. My expectation is that Tasmota gets better chances of managing the display if you do not leave out such pins, but I've not doing much testing/experimentation either way.

**Reply by RichardLaxton** [reply]
Display reset is connected to GPIO32 but I set it to Reset rather than Display Rst. I changed it as you suggested but it did not seem to make any difference. Thanks for the pickup though, the documentation that I was looking at used "Reset" as the reset pin.

**Reply by RichardLaxton** [reply]
Actually, setting the GPIO32 from Reset to Display Rst did make a difference. It broke my ability to fix the issue by swapping display types, I am guessing because the initialisation script is doing the same reset that the hard reset was previously doing.

I am now leaning towards this being some sort of inconsistency in the way that the screen is being initialised. Looks like I have to dig deep into the driver, initialisation process and the datasheet for this particular screen. Joy :-)

**Reply by sfromis** [reply]
As I have zero problems with ILI9341 displays and tasmota32-lvgl, including what looks like the board you have, I strongly doubt some general issue only fixable by code diving

**Comment by RichardLaxton** [comment]
OK, so I rewired everything and completely ignored the standard pins for SPI on an ESP32 and everything worked perfectly, as well as being a lot neater to wire up.

Closing this since it is obviously not a real issue.
